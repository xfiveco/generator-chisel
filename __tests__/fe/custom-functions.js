const fs = require('fs-extra');
const { defaultAnswers } = require('./helpers');

describe('FE Custom Functions', () => {
  test('Custom functions work ', async () => {
    await global.chiselTestHelpers.generateProjectWithAnswers(
      ['create'],
      defaultAnswers,
    );

    fs.outputFileSync(
      './src/templates/page-1.twig',
      /* HTML */ `
        <!DOCTYPE html>
        <ul>
          <li>{{ simple() }}</li>
          <li>{{ add(10, 20) }}</li>
          <li>{{ somethingAsync() }}</li>
        </ul>
      `,
    );

    fs.outputFileSync(
      './chisel.config.js',
      fs.readFileSync('./chisel.config.js', 'utf8').replace(
        /(?=};\n$)/,
        `
          staticFrontend: {
            functions: {
              simple({ functions: { revisionedPath } }) {
                return revisionedPath('scripts/app.js');
              },

              add(_, a, b) {
                return 'ADD: ' + (a + b);
              },

              async somethingAsync() {
                await new Promise(resolve => setTimeout(resolve, 1000));
                return 'After timeout';
              },
            },
          },
        `,
      ),
    );

    await global.chiselTestHelpers.runChiselScript(['build']);

    global.chiselTestHelpers.fileMatchesSnapshot('./dist/page-1.html');
  });
});
