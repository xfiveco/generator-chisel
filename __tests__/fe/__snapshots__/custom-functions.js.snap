// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FE Custom Functions Custom functions work  1`] = `
"<!DOCTYPE html>
<ul>
  <li>scripts/app.--HASH--.js</li>
  <li>ADD: 30</li>
  <li>After timeout</li>
</ul>"
`;
